File: java/strings/Reverse_Only_Letters_917.java
package java.strings;

public class Reverse_Only_Letters_917 {
        public String reverseOnlyLetters(String s) {
        int left=0;
        int right=s.length()-1;
        char[] s_chars=s.toCharArray();
        while (left<right){
            if (!Character.isAlphabetic(s_chars[left])){
                left+=1;
                continue;
            }
            if(!Character.isAlphabetic(s_chars[right])){
                right-=1;
                continue;
            }
            char temp=s_chars[left];
            s_chars[left]=s_chars[right];
            s_chars[right]=temp;
            left++;
            right--;

        }
        return new String(s_chars);
    }
}

File: python/strings/Reverse_Only_Letters_917.py
class Solution:
    def reverseOnlyLetters(self, s: str) -> str:
        s_lst=list(s)
        left=0
        right=len(s)-1
        while left<right:
            if not s_lst[left].isalpha():
                left+=1
                continue
            if not s_lst[right].isalpha():
                right-=1
                continue
            s_lst[left],s_lst[right]=s_lst[right],s_lst[left]
            left+=1
            right-=1
        return "".join(s_lst)
